ansys/dpf/core/plotter.py

```diff
@@ -14,6 +14,150 @@
+        except:
+            has_attribute_scalar_bar = False
+
+        if not has_attribute_scalar_bar:
+            kwargs.setdefault("stitle", "Mesh")
+        else:
+            if self._plotter.scalar_bar.GetTitle() is None:
+                kwargs.setdefault("stitle", "Mesh")
+        kwargs.setdefault("show_edges", True)
+        kwargs.setdefault("nan_color", "grey")
+        self._plotter.add_mesh(meshed_region.grid, **kwargs)
+
+    def add_field(self, field, meshed_region=None, **kwargs):
+        name = field.name.split("_")[0]
+        kwargs.setdefault("stitle", name)
+        kwargs.setdefault("show_edges", True)
+        kwargs.setdefault("nan_color", "grey")
+
+        # get the meshed region location
+        if meshed_region is None:
+            meshed_region = field.meshed_region
+        location = field.location
+        if location == locations.nodal:
+            mesh_location = meshed_region.nodes
+        elif location == locations.elemental:
+            mesh_location = meshed_region.elements
+        else:
+            raise ValueError(
+                "Only elemental or nodal location are supported for plotting."
+            )
+        component_count = field.component_count
+        if component_count > 1:
+            overall_data = np.full((len(mesh_location), component_count), np.nan)
+        else:
+            overall_data = np.full(len(mesh_location), np.nan)
+        ind, mask = mesh_location.map_scoping(field.scoping)
+        overall_data[ind] = field.data[mask]
+
+        # plot
+        self._plotter.add_mesh(meshed_region.grid, scalars=overall_data, **kwargs)
+
+    def show_figure(self, **kwargs):
+        background = kwargs.pop("background", None)
+        if background is not None:
+            self._plotter.set_background(background)
+
+        # show result
+        show_axes = kwargs.pop("show_axes", None)
+        if show_axes:
+            self._plotter.add_axes()
+        return self._plotter.show()
+
+
+class DpfPlotter:
+    def __init__(self, **kwargs):
+        self._internal_plotter = _InternalPlotter(**kwargs)
+
+    def add_mesh(self, meshed_region, **kwargs):
+        """Add a mesh to plot.
+
+        Parameters
+        ----------
+        meshed_region : MeshedRegion
+            MeshedRegion to plot.
+
+        Examples
+        --------
+        >>> from ansys.dpf import core as dpf
+        >>> from ansys.dpf.core import examples
+        >>> model = dpf.Model(examples.multishells_rst)
+        >>> mesh = model.metadata.meshed_region
+        >>> from ansys.dpf.core.plotter import DpfPlotter
+        >>> pl = DpfPlotter()
+        >>> pl.add_mesh(mesh)
+
+        """
+        self._internal_plotter.add_mesh(meshed_region=meshed_region, **kwargs)
+
+    def add_field(self, field, meshed_region=None, **kwargs):
+        """Add a field containing data to the plotter.
+        A meshed_region to plot on can be added.
```

Comment:
Leave a space between the short summary and the long summary to get this rendered correctly in Sphinx.
```suggestion

        A meshed_region to plot on can be added.
```